### ISSUE TITLE
Ohhh BrickStore's too smart on item ID change?

### ISSUE THREAD

**Issue by paramecie** [rgriebl/brickstore#295]
I'm puzzled.

I edited an XML using an "old" reference of a part (it changed 2 days ago).

Reference FTFpb237c01 - while the new is "FTG" not "FTF".

![image](https://user-images.githubusercontent.com/32078594/110722260-42ff9380-8212-11eb-9817-43cfd2bf0af9.png)

My problem is that BrickStore is too smart!

When you open the file it automatically changes the improper reference to the new one!

![image](https://user-images.githubusercontent.com/32078594/110721738-6118c400-8211-11eb-86ff-46f477f23191.png)

That's nice, but...

1) How does it know - you've the catalog log changes in the db, or it keep the previous one?
How long will it last - 1 change, 2 changes, all changes?

2) I don't agree that it silently changes a item. If it is wrong?
As BOTH ref and description are changed, I've no idea what it was before.
Problem is: how can I know as it silently replaces everything?

3) It's changed but NOT modified. So I can't save it, as it's not edited!

In short, niiiice feature but:

* **BrickStore should NEVER change anything in a file without the user interaction / agreement, IMO**

* **if something is changed, then it must be set "Modified"**

Million thanks!

**Comment by rgriebl** [comment]
Marking the document modified is not necessary. The changelog is simply re-applied on the next load and it goes back to AD 0 (Anno Dan) ;)
I had a message box in there before telling you it updated %n lots, but in the end, it wasn't necessary. And you - of all people - should understand the value of reducing the number of message boxes ;)

Plus, it also adheres to your mantra of not marking a file modified, unless you manually edit something in there.

**Comment by paramecie** [comment]
Strooongly try to make you think at it differently.

A BrickStore file is also a proof or a reference.
- My client sent me this file with an 1234 part, this is the file - but it shows 3456 now :-(
- I uploaded this stock, but ignore the different listing that I sent you, because it changes automatically, erm...
- I was sure there was a 70555pb767 in this file, but the Filter shows me no result, am I crazy!?

...

Maybe those are extreme examples, but a local document that automatically updates its content is to me a concept with important side effects.

Why not, instead:
* message box "3 references have been updated to the very recent catalog entries" => OK button only
* mark it modified

BOOM, done!

This way:
* user knows
* he has ONE click (for any number of changes), and 1 optional CTRL+S to *save* this change, if he wishes.

And if you don't change the BS file, then a grep or a search in the content of hundred of BS files (I do it often), will NOT bring you a result coherent with what shows BrickStore!

So you'll have an XML with FTFpb237c01, that shows FTGpb237c01; this isn't expected with an open format (XML) - IMHO.
